### ISSUE TITLE
Buffer overflow when using optimization flags

### ISSUE THREAD

**Issue by wietsefranssen** [UW-Hydro/VIC#311]
When you compile VIC with optimization enabled (`CFFLAG -O1` or higher) in stead of `-O0`, VIC will not run.
It throws the following messae after the vic-info text:
```*** buffer overflow detected ***```

note: I only tested this in VIC_image mode.

**Comment by jhamman** [comment]
Can you post the compiler and platform specifications that reproduce this behavior?

**Comment by wietsefranssen** [comment]
```
mpicc for MPICH version 3.0.4
Using built-in specs.
COLLECT_GCC=cc
COLLECT_LTO_WRAPPER=/usr/lib/gcc/x86_64-linux-gnu/4.8/lto-wrapper
Target: x86_64-linux-gnu
Configured with: ../src/configure -v --with-pkgversion='Ubuntu 4.8.4-2ubuntu1~14.04' --with-bugurl=file:///usr/share/doc/gcc-4.8/README.Bugs --enable-languages=c,c++,java,go,d,fortran,objc,obj-c++ --prefix=/usr --program-suffix=-4.8 --enable-shared --enable-linker-build-id --libexecdir=/usr/lib --without-included-gettext --enable-threads=posix --with-gxx-include-dir=/usr/include/c++/4.8 --libdir=/usr/lib --enable-nls --with-sysroot=/ --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --enable-gnu-unique-object --disable-libmudflap --enable-plugin --with-system-zlib --disable-browser-plugin --enable-java-awt=gtk --enable-gtk-cairo --with-java-home=/usr/lib/jvm/java-1.5.0-gcj-4.8-amd64/jre --enable-java-home --with-jvm-root-dir=/usr/lib/jvm/java-1.5.0-gcj-4.8-amd64 --with-jvm-jar-dir=/usr/lib/jvm-exports/java-1.5.0-gcj-4.8-amd64 --with-arch-directory=amd64 --with-ecj-jar=/usr/share/java/eclipse-ecj.jar --enable-objc-gc --enable-multiarch --disable-werror --with-arch-32=i686 --with-abi=m64 --with-multilib-list=m32,m64,mx32 --with-tune=generic --enable-checking=release --build=x86_64-linux-gnu --host=x86_64-linux-gnu --target=x86_64-linux-gnu
Thread model: posix
gcc version 4.8.4 (Ubuntu 4.8.4-2ubuntu1~14.04) 
```

**Comment by wietsefranssen** [comment]
`CFLAGS  =  ${INCLUDES} -ggdb -O0 -Wall -Wextra -std=c99 -DLOG_LVL=$(LOG_LVL)
`

**Comment by jhamman** [comment]
@wietsefranssen - I'm having trouble reproducing this error.  I'm wondering if you can take a first swing at this and get back to me with any ideas of what/where the bug is.  Thanks.

**Comment by wietsefranssen** [comment]
After a little investigation....
It goes wrong at:
https://github.com/UW-Hydro/VIC/blob/b20d6db73a982915263fd8491eaef8b0dd083852/vic/drivers/shared/src/output_list_utils.c#L207-L209
`OUT_MELT_ENERGY_BAND` goes over the limit of 20 (see https://github.com/UW-Hydro/VIC/blob/b20d6db73a982915263fd8491eaef8b0dd083852/vic/drivers/shared/include/vic_driver_shared.h#L404-L406)
Changing `char varname[20];` to eg `char varname[40];` solves the error.

I think this is a small but important thing (it will also overflow without optimisation flag).

What do you think? Should we go for safer solutions and avoid strcpy (and use for eg strlcpy) or just make the size of `char varname[];` bigger?

**Comment by wietsefranssen** [comment]
btw. I created a pull request with this little fix (#320). Just let me know if I am creating a bit too often pull requests now.

**Comment by jhamman** [comment]
small PRs are good.  Thanks for looking into this and for the fix.
